web/src/features/Dashboard/components/Dashboard.tsx

```diff
@@ -513,10 +517,25 @@ const Dashboard = (props: Props) => {
             selectedAppClusterDashboardResponse.prometheusAddress,
           metrics: selectedAppClusterDashboardResponse.metrics,
         },
+        lastUpdated: 0,
+        lastUpdatedDate: new Date(),
       });
     }
   }, [selectedAppClusterDashboardResponse]);
 
+  useEffect(() => {
+    const interval = setInterval(() => {
```

Comment:
this interval checks every second for the time that has passed since appStatus is updated.

Reply:
this is probably fine. 

- another way to do it would be to use `setState` in an interval that just updates the current time. (make sure to set your date with a function like [this](https://eight-bites.blog/en/2021/05/setinterval-setstate/) to avoid stale state: 

```
import React, { useEffect, useState } from 'react'

export const Sample = () => {
  let timer
  const [count, setCount] = useState(0)
  
  const updateCount = () => {
    timer = !timer && setInterval(() => {
      console.log('ticking')
+     setCount(prevCount => prevCount + 1) // new
    }, 1000)
  }
  
  useEffect(() => {
    updateCount()
    
    return () => clearInterval(timer)
  }, [])
  
  
  return <div>count is: {count}</div>
}
```

- the fetch handler could set the fetch time and the current time. 

- then you could have a `useEffect` that's watching both and updates the display string when they change. that way all of the logic is in taht effect and the jsx is just rendering the string.

Reply:
ah nevermind. that doesn't make sense. i think the link has an example of how you could do it with one use effect. either way this is fine.